### PR TITLE
Added Implementation of NAND, OR ,XNOR and NOT gates in python

### PR DESCRIPTION
**Pull request by karthiks2611**
### Describe your change:

* [x] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### Checklist:
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [ ] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [ ] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [ ] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

File: boolean_algebra/nand_gate.py [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    |    1    |    0    |    1   |
+    |    1    |    1    |    0   |
+    ------------------------------
+Refer - https://www.geeksforgeeks.org/logic-gates-in-python/
+"""
+
+
+def nand_gate(input_1: int, input_2: int) -> int:
+    """
+    Calculate AND of the input values
```
Comment (arjitarora26) [inline boolean_algebra/nand_gate.py:20]: AND -> NAND

File: boolean_algebra/nand_gate.py [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    >>> nand_gate(0, 0)
+    1
+    >>> nand_gate(0, 1)
+    1
+    >>> nand_gate(1, 0)
+    1
+    >>> nand_gate(1, 1)
+    0
+    """
+    return int(not ((input_1, input_2).count(0) == 0))
```
Comment (arjitarora26) [inline boolean_algebra/nand_gate.py:30]: can directly write 
`return int( (input_1, input_2).count(0) != 0)`

File: boolean_algebra/nand_gate.py (hunk 1 above)
Comment (karthiks2611) [reply boolean_algebra/nand_gate.py]: thanks for pointing it out. Will do the corrections

**Review by cclauss — APPROVED** [review]

**Comment by cclauss** [comment]
Added labels!!

**Pull request merged**